src/components/eip4626/user/MintAssetToken.tsx

```diff
@@ -0,0 +1,171 @@
+    mutateAsync: mint,
+    error: mintError,
+    isPending: isMintPending,
+  } = useWriteHtsTokenMint();
+
+  const { data: tokenBalance, error: tokenBalanceError } = useReadBalanceOf(
+    deployedHtsTokensAddress as EvmAddress,
+  );
+
+  const {
+    data: accountTokens,
+    isFetching,
+    hasNextPage,
+    fetchNextPage,
+  } = useAccountTokens();
+
+  useEffect(() => {
+    if (hasNextPage && !isFetching) {
+      fetchNextPage();
+    }
+  }, [accountTokens, fetchNextPage, hasNextPage, isFetching]);
+
+  const [tokenHasAssociation, setTokenHasAssociation] = useState(false);
+  useEffect(() => {
+    if (deployedHtsTokensAddress && accountTokens) {
+      let deployedTokenLowercase = deployedHtsTokensAddress.toLowerCase();
+      let result = accountTokens.some((token) => {
+        if (!token) return false;
+        let tokenSolidityAddress = AccountId.fromString(
+          token.token_id,
+        ).toSolidityAddress();
+
+        return `0x${tokenSolidityAddress}` === deployedTokenLowercase;
+      });
+
+      setTokenHasAssociation(result);
+    }
+  }, [accountTokens, deployedHtsTokensAddress, setTokenHasAssociation]);
+
+  return (
+    <>
+      {deployedHtsTokensAddress && (
+        <>
+          <Text>HTS Token CA: {deployedHtsTokensAddress}</Text>
+          <Text>HTS Token Proxy CA: {vaultAssetSelected}</Text>
+          <Text>
+            User balance of token:{" "}
+            {`${formatBalance(tokenBalance, vaultAssetSelectedDecimals)}`}
+          </Text>
+          <Button
+            isLoading={isAssociatePending}
+            isDisabled={tokenHasAssociation}
+            onClick={() =>
+              associate(
+                {
+                  tokenAddress:
+                    deployedHtsTokensAddress?.toString() as EvmAddress,
+                },
+                {
+                  onSuccess: () => {
+                    queryClient.invalidateQueries({
+                      queryKey: [QueryKeys.ReadAccountTokens],
+                    });
+                  },
+                },
+              )
+            }
+          >
+            {tokenHasAssociation ? `Token already associated` : `Associate`}
+          </Button>
+          <Button
+            isLoading={isMintPending}
+            onClick={() =>
+              mint(
+                {
+                  tokenAddress: vaultAssetSelected,
+                  mintAmount: formatNumberToBigint(
+                    DEFAULT_TOKEN_MINT_AMOUNT,
+                    vaultAssetSelectedDecimals,
+                  ),
+                },
+                {
+                  onSuccess: () => {
+                    queryClient.invalidateQueries({
+                      queryKey: [QueryKeys.ReadBalanceOf],
+                    });
+                  },
+                },
+              )
+            }
+          >
+            Mint {DEFAULT_TOKEN_MINT_AMOUNT} tokens
+          </Button>
+        </>
+      )}
+      {associateResult && (
```

Comment:
I was thinking about solutions how do we mapping those template rendering conditions:
```
const renderAlertTemplates = () => {
    if (associateResult) {
        return <result />
    } else {
        return <error />
    }
}
```

Reply:
good point, ye, its definitely a time to have some shared compo for those.
feel free to make one, I can come up with something next time, dont wanna expand the scope of this PR for now